packages/react-ai/src/components/AIConversation/context/ControlsContext.tsx

```diff
@@ -1,20 +1,24 @@
 import React from 'react';
-import { ConversationInputContext } from './ConversationInputContext';
+import { ConversationInputContextProps } from './ConversationInputContext';
 import { SuggestedPrompt } from '../types';
 import { ConversationMessage } from '../../../types';
+import { AttachmentContextProps } from './AttachmentContext';
+import { ConversationDisplayText } from '../displayText';
 
 export interface ControlsContextProps {
   Form?: React.ComponentType<
     {
       handleSubmit: (e: React.FormEvent<HTMLFormElement>) => void;
       allowAttachments?: boolean;
       isLoading?: boolean;
-    } & Required<ConversationInputContext>
+      displayText: Required<ConversationDisplayText>;
```

Comment:
Will making `displayText` required break composability?

Reply:
Nope, because the composable Form control doesn't take any props. This is to override the form component (the Form control grabs all the data and stuff from contexts and passes it to this component).

Reply:
But it did make it kinda wonky, changing to just take an onValidate callback that handles the validation and state update